faults/faults.py

```diff
@@ -0,0 +1,131 @@
+import numpy as np
+
+
+## Fault models
+"""
+    Fault(name=None)
+
+The base class of fault models.
+"""
+class Fault:
+    def __init__(self, name=None):
+        self.name = name
+
+    def __repr__(self):
+        return f"Fault name: {self.name}"
+
+    def get(t, u):
+        raise NotImplementedError("`Fault` class needs `get`")
+
+    def __call__(self, *args, **kwargs):
+        return self.get(*args, **kwargs)
+
+"""
+    LoE(time=0, index=1, level=1.0)
+
+A fault class for loss of effectiveness (LoE).
+
+# Parameters
+time: from when LoE is applied
+index: actuator index to which LoE is applied
+level: effectiveness (e.g., level=1.0 means no fault)
+"""
+class LoE(Fault):
+    def __init__(self, time=0, index=1, level=1.0, name = "LoE"):
+        super().__init__(name=name)
+        self.time = time
+        self.index = index
+        self.level = level
+
+    def __repr__(self):
+        print(super().__repr__())
+        return f"time = {self.time}, index = {self.index}, level = {self.level}"
+
+    def get(self, t, u):
+        effectiveness = np.ones_like(u)
+        if t >= self.time:
+            effectiveness[self.index-1] = self.level
+        return u * effectiveness
```

Comment:
index는 헷갈리지 않게 zero-base로 하는게 어떨까요?

그리고 코드는 간단하게 다음과 같이 해도 괜찮을 것 같은데..
```suggestion
        if t >= self.time:
            u[self.index] *= self.level
        return u
```

Reply:
ok
(tmi: 사실 들어온 입력을 in-place 로 바꾸는걸 비선호하는 편이라 저렇게 쓰긴 함)

Reply:
@seong-hun 
코드를 돌려보니까 side effect 가 있군요 (0.2, 0.04가 됨).
개인적으로는 out-of-place 로 변경하는 방식을 유지하는게 낫겠다고 생각해서, 다시 원래대로 코드를 돌리겠습니다 (zero-based indexing 은 유지)
```
Fault name: LoE
time = 2, index = 1, level = 0.2
Actuator command: [1. 1. 1. 1. 1. 1.], time: 0.0
Actual input: [1. 1. 1. 1. 1. 1.]
Actuator command: [1. 1. 1. 1. 1. 1.], time: 1.0
Actual input: [1. 1. 1. 1. 1. 1.]
Actuator command: [1. 1. 1. 1. 1. 1.], time: 2.0
Actual input: [1.  0.2 1.  1.  1.  1. ]
Actuator command: [1.  0.2 1.  1.  1.  1. ], time: 3.0
Actual input: [1.   0.04 1.   1.   1.   1.  ]
Fault name: Float
time = 2, index = 1, level = 0.0
Actuator command: [1. 1. 1. 1. 1. 1.], time: 0.0
Actual input: [1. 1. 1. 1. 1. 1.]
Actuator command: [1. 1. 1. 1. 1. 1.], time: 1.0
Actual input: [1. 1. 1. 1. 1. 1.]
Actuator command: [1. 1. 1. 1. 1. 1.], time: 2.0
Actual input: [1. 0. 1. 1. 1. 1.]
Actuator command: [1. 0. 1. 1. 1. 1.], time: 3.0
Actual input: [1. 0. 1. 1. 1. 1.]
```

Reply:
Good~